backend/backend.go

```diff
@@ -43,7 +43,7 @@ type Backend interface {
 
 	// CreateOrchestrationInstance creates a new orchestration instance with a history event that
 	// wraps a ExecutionStarted event.
-	CreateOrchestrationInstance(context.Context, *HistoryEvent) error
+	CreateOrchestrationInstance(context.Context, *HistoryEvent, *protos.CreateInstanceOption) error
 
```

Comment:
Usually we try to use the options pattern for optional parameters, which I believe is a common practice in Go. That would also allow us to make this change in a non-breaking way. Have you considered this?

Reply:
Updated to use option pattern, but I have to use a wrap as directly copy value of the `*protos.OrchestrationIdReusePolicy` gives me a passing lock value warning.